### ISSUE TITLE
How long does it take to run 10K iterations?

### ISSUE THREAD

**Issue by speedyray** [roatienza/Deep-Learning-Experiments#12]
How long does it take to run 10K iterations on a 2.7 GHZ intel Core i5 processor Mac book?

I was at 150 iterations after so many hours? Is there a problem with code, system or I need a GPU?

**Comment by daryabiparva** [comment]
Did you get your answer?

**Comment by jrtmendoza** [comment]
You have to understand that deep neural network architectures demand
enormous amount of parallelism. Executing these types of workload on a CPU
is futile and unproductive. Even a simple model that runs for 30 minutes on
a GPU usually takes a day or so on CPUs.

**Comment by negodfre** [comment]
Nvidia Geforce 1060 GPU 6GB; I am to get to 1000 iterations in about 2 minutes.

**Comment by speedyray** [comment]
Is it the Nvidia GEFORCE GTX 1060 which costs $299? https://www.nvidia.com/en-us/geforce/products/10series/geforce-gtx-1060/
Please provide more details on how or a link to the project if you can 
Thanks

**Comment by negodfre** [comment]
Yeah, with 6GB of RAM. It mentions it's sold out atm. Amazon shows only a used option. I am not sure if they don't make it anymore or what. 

Also, if it's helpful I am using an AMD Ryzen 7 2700X 8-core Processor. 
While running the GAN training, I am at about 56% usage on GPU & 10% on CPU with about 8 GB of RAM being used.

I can post total elapsed time when it's complete. I'm at about 4000 iterations right now.

**Comment by speedyray** [comment]
I would prefer to be able to run 10, 000 iterations in about 2 - 5 minutes. Any processors out there that can do this in 2019? Please update. Thanks

**Comment by daryabiparva** [comment]
I have used google colab. It did well.

**Comment by speedyray** [comment]
It did well is too vague!!

Please provide exact details.
How many minutes did it take to run 10000 iterations? What GPUs?

Scientific and empirical evidence only.

We will test all reported results to prove its authenticity!!!!!!
We believe in proven facts!!!!

**Comment by daryabiparva** [comment]
Sorry.

It was couple of month ago. I do not remember the exact details. I think It
took about 1 hour or less to finish 10000 iterations. I do not have a GPU.
So google colab is my only chance.
Sorry If I could not provide exact details.

**Comment by speedyray** [comment]
Hi daryabiparva, 
please kindly run the experiment again now so we all know, Sir. 
google Colab is free to use, please. No need to say Sorry when you can use something for free and help others.

Best regards,
Thanks
